demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -677,36 +673,86 @@ static void otaAppCallback( OtaJobEvent_t event,
     }
 }
 
+jobMessageType_t getJobMessageType( const char * pTopicName,
+                                    uint16_t topicNameLength )
+{
+    uint16_t index = 0U;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    bool isMatch = false;
+    jobMessageType_t jobMessageIndex = jobMessageTypeMax;
+
+    /* Lookup table for OTA job message string. */
+    static const char * const pJobTopicFilters[ jobMessageTypeMax ] =
+    {
+        OTA_TOPIC_PREFIX OTA_TOPIC_JOBS "/$next/get/accepted",
+        OTA_TOPIC_PREFIX OTA_TOPIC_JOBS "/notify-next",
+    };
+
+    /* Match the input topic filter against the wild-card pattern of topics filters
+    * relevant for the OTA Update service to determine the type of topic filter. */
+    for( ; index < jobMessageTypeMax; index++ )
+    {
+        mqttStatus = MQTT_MatchTopic( pTopicName,
+                                      topicNameLength,
+                                      pJobTopicFilters[ index ],
+                                      strlen( pJobTopicFilters[ index ] ),
+                                      &isMatch );
+        assert( mqttStatus == MQTTSuccess );
+
+        if( isMatch )
+        {
+            jobMessageIndex = index;
+            break;
+        }
+    }
+
+    return jobMessageIndex;
+}
+
 /*-----------------------------------------------------------*/
 
 static void mqttJobCallback( MQTTContext_t * pContext,
                              MQTTPublishInfo_t * pPublishInfo )
 {
     OtaEventData_t * pData;
     OtaEventMsg_t eventMsg = { 0 };
+    jobMessageType_t jobMessageType = 0;
 
     assert( pPublishInfo != NULL );
     assert( pContext != NULL );
 
     ( void ) pContext;
 
-    LogInfo( ( "Received job message callback, size %ld.\n\n", pPublishInfo->payloadLength ) );
-
-    pData = otaEventBufferGet();
+    jobMessageType = getJobMessageType( pPublishInfo->pTopicName, pPublishInfo->topicNameLength );
```

Comment:
seems the handler code `mqttJobCallback` is same for both job message types. could we as well remove this ?

Reply:
This is basically to identify the two job topics and other topics , and extending it in future.

Reply:
Agree, we could then make this a generic helper function in future.